gobblin-metrics/src/test/java/gobblin/metrics/performance/MetricsUpdater.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.performance;
+
+import lombok.Builder;
+import lombok.Getter;
+
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import com.beust.jcommander.internal.Lists;
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+
+import gobblin.metrics.MetricContext;
+import gobblin.metrics.RootMetricContext;
+
+
+/**
+ * A {@link Runnable} that updates a set of metrics for every call to {@link #run}.
+ *
+ * <p>
+ *   This class will automatically generate a metric context tree and a set of metrics at the lowest level context.
+ *   An instance is created through a builder with the following optional parameters:
+ *   * depth: number of levels in generated {@link MetricContext} tree. Default: 0
+ *   * context: base {@link MetricContext} used as the root of the generated tree. Default: {@link RootMetricContext}.
+ *   * counters: number of counters to generate. Named counter0, counter1, ...
+ *   * meters: number of meters to generate. Named meter0, meter1, ...
+ *   * histograms: number of histograms to generate. Named histogram0, histogram1, ...
+ *   * timers: number of timers to generate. Named timer0, timer1, ...
+ * </p>
+ */
+public class MetricsUpdater implements Runnable {
```

Comment:
Would recommend using `Callable` instead of `Runnable` as `Callable` has better handing of exceptions thrown in the `run()` method.